src/detect-lua.c

```diff
@@ -1396,14 +1396,12 @@ static int LuaMatchTest01a(void)
     StreamTcpInitConfig(true);
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NOT_NULL(de_ctx);
     de_ctx->flags |= DE_QUIET;
 
     s = DetectEngineAppendSig(de_ctx, sig);
     if (s == NULL) {
-        printf("sig parse failed: ");
+        printf("sig parse failed: "); //
         goto end;
     }
```

Comment:
This `if` should be converted to a `FAIL_IF_NULL` macro. When that's done, this `printf` becomes obsolete, so could be removed.

Reply:
This applies to similar situations where if statements are about comparisons with `NULL`